cmake/FindJavacc.cmake

```diff
@@ -4,6 +4,20 @@ mark_as_advanced(JAVACC_EXECUTABLE)
 if(JAVACC_EXECUTABLE)
   set(JAVACC_FOUND 1)
   message(STATUS "The javacc executable: ${JAVACC_EXECUTABLE}")
+  if(JAVACC_EXECUTABLE)
+    execute_process(
+      COMMAND "${JAVACC_EXECUTABLE}" --version
+      OUTPUT_VARIABLE JAVACC_TEMP_VERSION
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+      RESULT_VARIABLE _JavaCC_version_result
+    )
+    if(_JavaCC_version_result)
+      message(WARNING "Unable to determine JavaCC version: ${_JavaCC_version_result}")
```

Comment:
I think this `if` is always true because `javacc --version` returns 1 thus `RESULT_VARIABLE _JavaCC_version_result` sets 1.
Then this massage shows:

```
$ cmake ../
-- The javacc executable: /home/wsh/qjava/javacc/scripts/javacc
CMake Warning at cmake/FindJavacc.cmake:15 (message):
  Unable to determine JavaCC version: 1
Call Stack (most recent call first):
  vhdlparser/CMakeLists.txt:1 (find_package)
```

How about this one?

```cmake
find_program(JAVACC_EXECUTABLE NAMES javacc javaCC Javacc JavaCC javacc.bat DOC "path to the javacc executable")
mark_as_advanced(JAVACC_EXECUTABLE)
if(JAVACC_EXECUTABLE)
  set(JAVACC_FOUND 1)
  message(STATUS "The javacc executable: ${JAVACC_EXECUTABLE}")
  if(JAVACC_EXECUTABLE)
    execute_process(
      COMMAND "${JAVACC_EXECUTABLE}" --version
      OUTPUT_VARIABLE JAVACC_TEMP_VERSION
    )
    string(REGEX MATCH ".* ([0-9]+\.[0-9]+[\.0-9]*) .*" JAVACC_TEMP_VERSION2_UNUSED ${JAVACC_TEMP_VERSION})
    if(CMAKE_MATCH_1)
      set(JAVACC_VERSION ${CMAKE_MATCH_1})
    else()
      message(STATUS "unexpected output from `javacc --version`:\n${JAVACC_TEMP_VERSION}")
      message(STATUS "abort using ${JAVACC_EXECUTABLE}, using existing files")
      set(JAVACC_FOUND 0)
    endif()
  endif()
else()
  set(JAVACC_FOUND 0)
  message(STATUS "The javacc executable not found, using existing files")
endif()
```

with JavaCC 5.0:

```sh
$ cmake ../
-- The javacc executable: /usr/bin/javacc
--   Doxygen requires at least JavaCC version 7.0.5 (installed: 5.0)
--   Fall back to JavaCC not installed.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/wsh/qc/doxygen/build
```

If `javacc --version` invoked `cowsay hi there`:

```sh
$ cmake ../
-- The javacc executable: /usr/bin/javacc
-- unexpected output from `javacc --version`:
 __________
< hi there >
 ----------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

-- abort using /usr/bin/javacc, using existing files
-- Configuring done
-- Generating done
-- Build files have been written to: /home/wsh/qc/doxygen/build
```

Reply:
(By the way, I prefer `if(NOT ...) return()` pattern.)

```cmake
find_program(JAVACC_EXECUTABLE NAMES javacc javaCC Javacc JavaCC javacc.bat DOC "path to the javacc executable")
mark_as_advanced(JAVACC_EXECUTABLE)
if(NOT JAVACC_EXECUTABLE)
  set(JAVACC_FOUND 0)
  message(STATUS "The javacc executable not found, using existing files")
  return()
endif()
set(JAVACC_FOUND 1)

execute_process(
  COMMAND "${JAVACC_EXECUTABLE}" --version
  OUTPUT_VARIABLE JAVACC_TEMP_VERSION
)
string(REGEX MATCH ".* ([0-9]+\.[0-9]+[\.0-9]*) .*" JAVACC_TEMP_VERSION2_UNUSED ${JAVACC_TEMP_VERSION})
set(JAVACC_TEMP_VERSION_MATCH ${CMAKE_MATCH_1})
if(NOT JAVACC_TEMP_VERSION_MATCH)
  message(STATUS "unexpected output from `javacc --version`:\n${JAVACC_TEMP_VERSION}")
  message(STATUS "abort using ${JAVACC_EXECUTABLE}, using existing files")
  set(JAVACC_FOUND 0)
  return()
endif()
set(JAVACC_VERSION ${JAVACC_TEMP_VERSION_MATCH})

message(STATUS "The javacc executable: ${JAVACC_EXECUTABLE} (version: ${JAVACC_TEMP_VERSION_MATCH})")
```

Reply:
- I didn't get the exit code 1 as I use a small wrapper script and this does not return the exit codes.
- the `REGEX MATCH` is nice, probably better than my construct
- the usage of `NOT` is a matter of taste, but I'll consider it.